### PR TITLE
Feat / migrate radar to 0x v3

### PR DESCRIPTION
**Pull request by kmcgowan2000**
**Before submitting this PR, please make sure**:
- [x] Your code builds clean without any errors or warnings
- [x] You are using approved title ("feat/", "fix/", "docs/", "refactor/", etc)

**Optional**:
- Related Github issue: https://github.com/CoinAlpha/hummingbot/issues/1229
- Related Clubhouse Story: 6791

**A description of the changes proposed in the pull request**:
Migrate the Radar Relay connector to v3. Includes updating some dependencies.

### CODE REVIEW

File: hummingbot/wallet/ethereum/watcher/new_blocks_watcher.py [hunk 1]
```diff
@@ -107,7 +107,7 @@ def block_number(self) -> int:
                         incoming_block: AttributeDict = await self.call_async(
                             functools.partial(
                                 self._w3.eth.getBlock,
-                                self._block_number_to_fetch,
+                                'latest',
```
Comment (chowryan) [inline hummingbot/wallet/ethereum/watcher/new_blocks_watcher.py]: if we fetch only the latest block theres a chance we might skip blocks?

File: setup/environment-linux.yml [hunk 2]
```diff
@@ -137,9 +137,9 @@ dependencies:
     - toolz==0.9.0
     - twisted==18.9.0
     - txaio==18.8.1
-    - typing-extensions==3.7.2
+    - typing-extensions==3.7.4.1
     - virtualenv==16.7.6
-    - web3==4.8.3
+    - web3==5.5.1
```
Comment (chowryan) [inline setup/environment-linux.yml]: this is a major version bump. I think there might be deprecated code we might need to fix? Do we need to do this now?

**Comment by chowryan** [comment]
we should be able to delete the old 0x helper and exchange files too

File: setup/environment-linux.yml (hunk 2 above)
Comment (kmcgowan2000) [reply setup/environment-linux.yml]: I think fixed all now?

File: hummingbot/wallet/ethereum/watcher/new_blocks_watcher.py (hunk 1 above)
Comment (kmcgowan2000) [reply hummingbot/wallet/ethereum/watcher/new_blocks_watcher.py]: fixed.

**Pull request merged**